File: bmi.c
// Include header file for prototypes of scanf and printf
#include <stdio.h>

// Prototype of calcBmi function
float calcBmi(float h,float w);

// Main function - entry point
int main(){
    // Local variable declaration
    float height;
    float weight;
    float bmi;

    // Ask user for height in cms
    printf("Enter your height in cms: ");
    scanf("%f",&height);

    // Ask user for weight in kgs
    printf("Enter your weight in kgs: ");
    scanf("%f",&weight);

    // Call calcBmi function
    bmi=calcBmi(height,weight);

    // Print result
    printf("Your body mass index is %.2f kg/cm^2",bmi);

    if(bmi<=18.5){
        printf("\nBajo peso");    
        }
    else if (bmi>18.5 && bmi<=24.9){
        printf("\nPeso normal");
    }
    else if (bmi>24.9 && bmi<29.9){
        printf("\nSobrepeso");
    }
    else if(bmi>=29.9){
        printf("\nObesidad");
    }
}

// Implementation of calcBmi function
float calcBmi(float h,float w){
    float bmi;

    // Compute body mass index
    bmi=w/((h/100.0)*(h/100.0));
    return bmi;
}

File: bmi.py
# Función para calcular el índice de masa corporal (IMC)
def calc_bmi(height, weight):
    bmi = weight / ((height / 100.0) ** 2)
    return bmi

# Función principal
def main():
    # Solicitar al usuario la altura en centímetros
    height = float(input("Ingrese su altura en cms: "))

    # Solicitar al usuario el peso en kilogramos
    weight = float(input("Ingrese su peso en kgs: "))

    # Calcular el IMC llamando a la función calc_bmi
    bmi = calc_bmi(height, weight)

    # Imprimir el resultado
    print(f"Su índice de masa corporal es {bmi:.2f} kg/m^2")
    
    if bmi<=18.5:
        print("\nBajo peso")
    
    elif bmi>18.5 and bmi<=24.9:
        print("\nPeso normal")
 
    elif bmi>24.9 and bmi<29.9:
        print("\nsobrepeso")
        
    elif bmi>=29.9:
        print("\nobesidad")    
    
        
# Ejecutar la función principal
if __name__ == "__main__":
    main()
